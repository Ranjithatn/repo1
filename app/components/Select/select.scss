

.select.no-arrow select {
  -webkit-appearance: none;
  appearance: none;
}


