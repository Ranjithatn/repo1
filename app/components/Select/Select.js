import React from 'react'
import Icon from "../Icon/Icon";
import "./select.scss";

const Select = ({
  className,
  name,
  leftIcon,
  id,
  options,
  value,
  onChange,
  onClick,
  formatMessage,
  defaultValue,
  style={},
  disabled=false,
  priorityDefault
}) => {

   return (
    <div className={"select " + (className ? " " + className : "")}>
     {leftIcon && <Icon icon={leftIcon} onClick={onClick}/>}
      <select name={name ? name : ""} id={id ? id : ""} value={value} defaultValue={defaultValue} onChange={onChange} disabled={disabled} style={style}>
        {!priorityDefault&&
        <option  value="" selected ></option>}
        {
          options && Array.isArray(options) && options.length && options.map(option => {
            return (
              <option key={option.value} value={option.value}> {formatMessage({id: option.displayName})}</option>
            )
          })
        }
      </select>

    </div>
   )
}

export default Select;