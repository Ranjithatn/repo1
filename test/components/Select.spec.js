import React from "react";
import toJson from "enzyme-to-json";
import Select from "../../app/components/Select/Select";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe( '<Select />', () => {

	it("should render without crashing", () => {
		const component = mount(
			<Select />
		);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});


	it("has correct className prop", () => {
		const props = { className: 'hello-world-select' }
		const component = shallow(
			<Select {...props} />
		);
		expect( component.props().className ).toBe('select  hello-world-select');
	});


	it("renders the component with all the given props", () => {
		const props = {
			className: 'hello-world-select',
			id: '_user_select',
			name: 'select_user',
			value: 'John Doe',
			options: [
				{ value: 'Jane Smith', displayName: 'Jane Smith', },
				{ value: 'John Doe', displayName: 'John Doe', },
				{ value: 'Gintama', displayName: 'Gintama', },
			],
			formatMessage: (data) => data,
		}
		const component = shallow(
			<Select {...props} />
		);

		expect( component.find('select').length ).toBe(1);
		expect( component.find('select').props().value ).toBe(props.value);
		expect( component.find('option').length ).toBe(4); // 4 because 1 empty option is added by default
	});

});

