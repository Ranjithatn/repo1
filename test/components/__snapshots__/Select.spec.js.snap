// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render without crashing 1`] = `
<Select>
  <div
    className="select "
  >
    <select
      disabled={false}
      id=""
      name=""
      style={Object {}}
    >
      <option
        selected={true}
        value=""
      />
    </select>
  </div>
</Select>
`;
